libs/prelude/Language/Reflection/Elab.idr

```diff
@@ -83,6 +83,34 @@ record FunDefn a where
   clauses : List (FunClause a)
 
 
+||| A constructor to be associated with a new datatype.
+record ConstructorDefn where
+  constructor Constructor
+
+  ||| The name of the constructor. The name must _not_ be qualified -
+  ||| that is, it should begin with the `UN` or `MN` constructors.
```

Comment:
And interface methods can be declared with a different namespace than their parent interface. So at the moment, it’s all uniform.